platform/linux-generic/arch/aarch64/odp_crypto_armv8.c

```diff
@@ -658,13 +999,78 @@ odp_crypto_session_create(const odp_crypto_session_param_t *param,
 			rc = -1;
 		}
 		break;
+	case ODP_AUTH_ALG_SHA1_HMAC:
+		/* Only AES-CBC and HMAC-SHA chained operation is supported */
+		if (param->cipher_alg != ODP_CIPHER_ALG_AES_CBC) {
+			rc = -2;
+			break;
+		}
+
+		if (param->cipher_alg == ODP_CIPHER_ALG_AES_CBC) {
```

Comment:
This condition is always true at this point, so testing it is unnecessary. The code in the body could be put in a function to remove clutter from the session creation function and to allow reuse between SHA-1 and SHA-2.